android/src/com/frostwire/android/gui/adapters/menu/ResumeDownloadMenuAction.java

```diff
@@ -58,4 +69,55 @@ protected void onClick(Context context) {
             }
         }
     }
+
+    @Override
+    public void onDialogClick(String tag, int which) {
+        if (tag.equals(DLG_TURN_BITTORRENT_BACK_ON)) {
+            if (which == Dialog.BUTTON_NEGATIVE) {
+                UIUtils.showLongMessage(getContext(),
+                        R.string.the_file_could_not_be_seeded_bittorrent_will_remain_disconnected);
+            } else if (which == Dialog.BUTTON_POSITIVE) {
+                onBittorrentConnect();
+            }
+        }
+    }
+    private void showBittorrentDisconnectedDialog() {
+        YesNoDialog dlg = YesNoDialog.newInstance(
+                DLG_TURN_BITTORRENT_BACK_ON,
+                R.string.bittorrent_off,
+                R.string.bittorrent_is_currently_disconnected_would_you_like_me_to_start_it_for_you,
+                YesNoDialog.FLAG_DISMISS_ON_OK_BEFORE_PERFORM_DIALOG_CLICK);
+        dlg.setOnDialogClickListener(this);
+        dlg.show(((Activity) getContext()).getFragmentManager());
+    }
+
+    private void onBittorrentConnect() {
+
+        if (ConfigurationManager.instance().getBoolean(Constants.PREF_KEY_NETWORK_BITTORRENT_ON_VPN_ONLY) &&
+                !NetworkManager.instance().isTunnelUp()) {
+            if (getContext() instanceof Activity) {
+                UIUtils.showShortMessage(((Activity) getContext()).getWindow().getDecorView().getRootView(), R.string.cannot_start_engine_without_vpn);
+            } else {
+                UIUtils.showShortMessage(getContext(), R.string.cannot_start_engine_without_vpn);
+            }
+        } else {
+            Engine.instance().getThreadPool().execute(new Runnable() {
```

Comment:
I think this will go away.

Reply:
ok, we'll have to do a little refactor since we'll now have some repeated code from `SeedAction.onBittorrentConnect()` which has a Context-leak free runnable.

Reply:
so in theory, this now will be better